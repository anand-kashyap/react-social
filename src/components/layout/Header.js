import React from 'react'
import { Link } from 'react-router-dom';
const Header = () => {
  return (
    <nav>
      <div className="nav-wrapper">
        <Link to="/" className="heading">SnappyLog</Link>
        <ul id="nav-mobile" className="right">
          <li>
            <Link to="/login" >Login</Link>
          </li>
        </ul>
      </div>
    </nav>
  )
}


export default Header;