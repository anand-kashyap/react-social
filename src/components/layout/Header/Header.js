import React from 'react'
import { Link } from 'react-router-dom';

import { navWrapper, heading } from './Header.module.scss';
const Header = () => {
  return (
    <nav>
      <div className={navWrapper}>
        <Link to="/" className={heading}>SnappyLog</Link>
        <ul id="nav-mobile" className="right">
          <li>
            <Link to="/login" >Login</Link>
          </li>
        </ul>
      </div>
    </nav>
  )
}


export default Header;