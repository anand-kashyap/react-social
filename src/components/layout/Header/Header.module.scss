.navWrapper {
  padding: 0 15px;
  .heading {
    position: absolute;
    color: #fff;
    display: inline-block;
    font-size: 2.1rem;
    padding: 0;
  }
}
