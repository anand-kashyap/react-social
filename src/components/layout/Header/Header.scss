nav {
  background-color: var(--base-color);
  a {
    color: var(--content-color);
  }
  .nav-wrapper {
    padding: 0 15px;
    .heading {
      position: absolute;
      color: #fff;
      display: inline-block;
      font-size: 2.1rem;
      padding: 0;
    }
    .dark-mode {
      button {
        height: inherit;
        line-height: inherit;
        color: inherit;
        &:hover {
          background-color: rgba(0, 0, 0, 0.1);
        }
        i {
          transform: rotate(45deg);
          // color: var(--text-color);
          line-height: inherit;
          height: inherit;
        }
      }
    }
  }
}
