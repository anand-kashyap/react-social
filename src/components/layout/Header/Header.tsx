import React, { useEffect, useState, useContext } from 'react';
import { Link, useHistory } from 'react-router-dom';
import authContext from 'context/auth/authContext';

import './Header.scss';
const Header = () => {
  const [darkMode, setDarkMode] = useState(false),
    hist = useHistory(),
    { isAuth, deleteToken } = useContext(authContext);

  useEffect(() => {
    if (darkMode) {
      document.body.setAttribute('data-theme', 'dark');
    } else {
      document.body.removeAttribute('data-theme');
    }
  }, [darkMode]);

  useEffect(() => {
    setDarkMode(window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches);
  }, []);

  const logOut = () => {
    deleteToken();
    hist.push('/login');
  }
  return (
    <nav>
      <div className='nav-wrapper'>
        <Link to='/' className='heading'>
          SnappyLog
        </Link>
        <ul id='nav-mobile' className='right'>
          <li className='dark-mode'>
            <button data-testid='dark-mode-btn'
              className='btn btn-small waves-effect btn-flat'
              onClick={() => setDarkMode(!darkMode)}
            >
              <i data-testid='dark-mode' className='material-icons-outlined'>
                {darkMode ? 'wb_sunny' : 'brightness_2'}
              </i>
            </button>
          </li>
          {isAuth ? <li data-testid='logout' className='logout-link'>
            <button
              className='btn btn-small waves-effect btn-flat'
              onClick={logOut}
            ><i className='material-icons-outlined'>exit_to_app</i>
            </button>
          </li>
            : <li className='login-link' data-testid='login'>
              <Link to='/login'>
                <i className='material-icons-outlined small'>perm_identity</i>
              </Link>
            </li>}
        </ul>
      </div>
    </nav>
  );
};

export default Header;
