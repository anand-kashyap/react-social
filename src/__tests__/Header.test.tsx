import React from 'react'
// import { toBeInTheDocument } from '@testing-library/jest-dom';
import { render, cleanup, fireEvent, screen } from '@testing-library/react'
import { Router } from 'react-router-dom'
import Header from 'components/layout/Header/Header';
import { createMemoryHistory } from 'history'
import AuthContext from 'context/auth/authContext';

const renderHeader = (props = {}, isAuth = false) => {
  const history = createMemoryHistory();
  return render(
    <AuthContext.Provider value={{ isAuth }}>
      <Router history={history}>
        <Header {...props} />
      </Router>
    </AuthContext.Provider>
  );
};

describe('Header Component', () => {
  afterEach(cleanup);
  // expect.extend({ toBeInTheDocument });

  it('should take a snapshot', () => {
    const { asFragment } = renderHeader();
    expect(asFragment()).toMatchSnapshot()
  });

  it('should toggle theme', () => {
    const { getByTestId } = renderHeader(),
      { textContent } = getByTestId('dark-mode'),
      btn = getByTestId('dark-mode-btn');

    fireEvent.click(btn);
    const { textContent: newText } = getByTestId('dark-mode');

    expect(newText).toBeTruthy();
    expect(newText).not.toBe(textContent);
  });

  it('should toggle theme 2 number of times', () => {
    const { getByTestId } = renderHeader(),
      btn = getByTestId('dark-mode-btn');
    btn.onclick = jest.fn();

    fireEvent.click(btn);
    fireEvent.click(btn);
    expect(btn.onclick).toHaveBeenCalledTimes(2);
  });

  it('should show login link if user is logged out', () => {
    const { getByTestId } = renderHeader(),
      loginLink = getByTestId('login'),
      logOutLink = screen.queryByTestId('logout');
    expect(loginLink).toBeInTheDocument();
    expect(logOutLink).not.toBeInTheDocument();
  });

  it('should show logout link if user is logged in', () => {
    const { getByTestId } = renderHeader({}, true),
      loginLink = screen.queryByTestId('login'),
      logOutLink = getByTestId('logout');
    expect(logOutLink).toBeInTheDocument();
    expect(loginLink).not.toBeInTheDocument();
  });

});