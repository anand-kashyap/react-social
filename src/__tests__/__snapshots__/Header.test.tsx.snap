// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should take a snapshot 1`] = `
<DocumentFragment>
  <nav>
    <div
      class="nav-wrapper"
    >
      <a
        class="heading"
        href="/"
      >
        SnappyLog
      </a>
      <ul
        class="right"
        id="nav-mobile"
      >
        <li
          class="dark-mode"
        >
          <button
            class="btn btn-small waves-effect btn-flat"
            data-testid="dark-mode-btn"
          >
            <i
              class="material-icons-outlined"
              data-testid="dark-mode"
            >
              wb_sunny
            </i>
          </button>
        </li>
        <li
          class="login-link"
          data-testid="login"
        >
          <a
            href="/login"
          >
            <i
              class="material-icons-outlined small"
            >
              perm_identity
            </i>
          </a>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
